### ISSUE TITLE
[Ubuntu] testresources needed for setuptools ?

### ISSUE THREAD

**Issue by mconsidine** [rlancaste/AstroPi3#22]
Hi,
I'm trying to get this setup on a Rock Pi 4 B Plus using an Ubuntu (focal) server image.  The script executes seemingly without a problem, until near the end where "setuptools" is getting installed.  At that point there is an error stating that "testresources" is not found.  It does not stop the script, though.

Upon reboot, all I have is a blank desktop with a status monitoring window and "Activities" as the only menu pick at the top of the screen.  The static IP address doesn't seem to have taken and the unit doesn't show up on the network.  KStars can be run but I don't see any obvious INDI manager.

Can you tell me if the "setuptools" issue is breaking something, as far as you know?
And can you point me to a blog posting or any discussion which would give me an idea of what I should expect to see happen upon rebooting after the script is run?

Many thanks - this looks like it will be great!
Regards,
Matt Considine

**Comment by rlancaste** [comment]
Hmm, this is very strange.  setup tools is just a package involving python. it shouldn't affect anything else, like bring down the system or making the desktop disappear.  That being said, if there is some problem with it, you can disable that part of the script without issue.  That part simply sets up indiweb, the indiwebmanagerapp, and things they require.  If the script is failing there, you might try running those steps separate from the script and see where it goes wrong. 

After rebooting, you should see the desktop as shown in the screenshots I posted to the repo.  https://github.com/rlancaste/AstroPi3

About the Rock Pi.   I had been really excited about the Rock64 when it first came out since there wasn't a 64 bit OS for raspberry pi and it didn't have USB3.  The Rock64 had both of those.  When mine arrived and I started playing with it, I ran into all kinds of problems with their system, especially related to the display compositor and VNC.  I seem to remember a couple of other issues, but I do remember those things made it very unusable, like making windows or screens go black or look unreadable.  My script was able to set it all up, but the other issues with their system made it unpleasant to use.  The RockPi 4 is not the Rock64, and it is very possible that they fixed all these issues I don't know.  I just thought I would let you know about that.

Thanks,

Rob

**Comment by mconsidine** [comment]
Hi Rob,
Thanks for the reply - and the great work on this.

I was able to get testresources via
  sudo apt install python3-testresources
but still found that indiwebmanagerapp wasn't getting found in the PPA.  It looks like the PPA only has a version for Ubuntu 18 and the version of Ubuntu I was pulling down (server version) was 20.04 (focal).

Also, I saw a warning flash by about the location for indi-web not being on the environment path.

So I found your git for it and can build that by first getting the dependencies.

So far, I have not been able to duplicate any of the desktops you show :)  With Ubuntu server I get the process monitor.  I'm going to load up the debian-xfce image and try the install there.

Ideally, I'd like to be able to power this on, go back into the house, VPN to it, and start playing.  I think that means I need a headless configuration.  Any tips to share on that?

This all "feels" to me like an issue or issues that will get resolved by getting various settings right, as opposed to something not working right.

Thanks for any help/tips.
Matt

**Comment by rlancaste** [comment]
hmm, I will need to address that with Jasem, I made the indiwebmanagerapp and I don't want it to not be available ;-)

Yes, the headless configuration is really what this script is designed for, but note when I mean headless, I still want a GUI to be running and accessible via VNC.  The goal is that when it is done running, samba is set up for file access, VNC is setup for VNC use, ssh is setup, it automatically logs in, the astronomy software is all set up, and you can use indiwebmanagerapp to manage the indiserver on the pi.

I didn't design the script to be a truly headless system because a great way to use the pi is to set it up to run KStars, Ekos and everything and just use your laptop to VNC in every now and then to check on things.  You can also use it as an INDI server and connect in with your laptop the way you are thinking of doing it.  The system the script sets up can be used both ways.  Also note, running PHD2 on the PI rather than guiding remotely with your laptop makes it run much faster and work better.

**Comment by rlancaste** [comment]
Ok I think indiwebmanagerapp is updated in the ppa now

**Comment by mconsidine** [comment]
Thanks.  I'm going to consider this issue closed as I don't really think it was related to your scripts.
As far as progress, I'm getting there.  Just can't seem to connect wirelessly yet.
Matt
![Screenshot_2022-03-20_17-38-10](https://user-images.githubusercontent.com/586812/159187925-94ae1df0-4f23-4c67-8e1d-c00f8a521720.png)

**Comment by rlancaste** [comment]
This looks pretty good.  For connecting wirelessly, as long as the pi is either connected to a network you are on, or the pi is hosting its own network and you joined it, it should connect and work fine remotely.  You can try to connect to the pi using the name or the IP address on the various services.  Are you saying VNC is not working right, INDI is not connecting, that you can't ping or ssh into it, or that you can't connect to it via samba?

**Comment by mconsidine** [comment]
This is not a Raspberry but a Rock Pi 4 B Plus.  The recent os images are
for a Debian Buster desktop and an Ubuntu Server (with no desktop
environment).

I got your scripts running on Debian, but it took some edits and I couldn't
connect to a hotspot.  So then I tried an Armbian image that was Ubuntu
based and with a desktop.  That seemed to go together without edits, but
still couldn't get to a hotspot (and then in trying to fix that I managed
to mess up the config files. My bad...)

I also tried starting with the Ubuntu server and then adding MATE, to get
to something your scripts expect.  But trying to install that via tasksel
never went anywhere.  Not sure why - tasksel has to be one of the least
informative programs I've ever seen.

So ... I can seem to either get a RockPi Debian or Armbian Focal XFCE image
to ultimately install.  And I can use x11vnc paired to tighten viewer to
see it on the network, when its wired.

Now I need to try getting to it unwired.  And maybe I need to roll back to
an older RockPi Ubuntu image (ie 18.04), which I think is what your Ubuntu
script was built around.

Clear as mud?  (It's mud season up here in Vermont ...)

Matt

**Comment by rlancaste** [comment]
Sure, that sounds good.   Really my scripts are intended to be modified depending on various setups and configurations.  It is meant to be customizable.  I assume when you tried Debian buster, you were using the setupDebianSBC script and as you were using the ubuntu serve one you were using the setupUbuntySBC script?  If you do find something that needs fixing please let me know.   I did run the scripts extensively when I was writing them, and worked to perfect them, but of course, things do change and break.

Thanks for the feedback!

**Comment by mconsidine** [comment]
Ok, to close this particular issue out ...
I have successfully used these scripts to install everything on a new Rock Pi 4 B Plus (4G/64G eMMC).  The unit I had did not (seem to) have an OS installed, so what I did was used Etcher
[https://www.balena.io/etcher/](url)
on a Windows machine to burn a Radxa-supplied OS image from github
[https://github.com/radxa/rock-pi-images-released/releases](url)
to a 16GB micro SD card.

Specifically, I burned the latest Debian image.  When I then booted into that on the Rock Pi, I went back to that site via the browser and pulled down the Ubuntu Server image
(specifically: 64-bit Ubuntu 20.04 "Focal" , [https://github.com/radxa/rock-pi-images-released/releases/download/v20210824/rockpi4b_ubuntu_focal_server_arm64_20210823_2154-gpt.img.gz](url) )
to the 16GB microSD I was using:
  (e.g [https://www.buycott.com/upc/804272745719](url) )

When that was burned (see Radxa wiki) to the eMMC of the Rock Pi and then hard-booted with a switch from the power strip, I got the familiar text login prompt.  For the Rock Pi the username and password at this stage is rock/rock.  At this point I can boot into Ubuntu Server like I wanted to *and* I have a backup bootable microSD.

I enabled the testing version of focal via an edit, ie
  `sudo nano /etc/apt/sources.list.d/apt-radxa-com.list`

I also enabled the GPIO pins via 
  `sudo nano /boot/hw_intfc.conf`
though I'm not sure that is strictly necessary for a basic install.  (Essentially, I wanted to avoid having to come back to do this later ...)
  
After updating and upgrading
  `sudo apt update`
  `sudo apt upgrade`

I created a new user via
  `sudo adduser astropi3`
and added this user to the same groups "rock" was part of, ie
  `sudo adduser astropi3 sudo`
  `sudo adduser astropi3 audio`
  `sudo adduser astropi3 video`

I have no idea if that was strictly necessary to add to audio and video, but I did it anyway.

Next, I added a lightweight desktop environment.  I chose Ubuntu Server because I wanted to use the latest OS image available, but the AstroPI3 Ubuntu script was created in a MATE desktop environment.  (And while I was able to get the Debian OS up and running, alot of things needed to get compiled from source it seemed when running the scripts.  So I went back to Ubuntu, which I am more familiar with anyway).

To add the MATE desktop I did
  `sudo apt install ubuntu-mate-core`
using this page as guidance
  [https://itsfoss.com/install-gui-ubuntu-server/](url)

So at this point I now have a more up to date Ubuntu image with a MATE desktop running on the Rock Pi.  Time to get AstroPi3...

I rebooted again and logged in as my new user, then brought up the terminal window to do this:
  `sudo apt install python3-testresources git wget`
(which was a double check to make sure I had those installed).

At this point I'm at steps 7/8 of the readme and everything goes "as advertised" :
  `git clone https://github.com/rlancaste/AstroPi3.git`
  `cd AstroPi3`

Hope this helps someone else.  I'm closing this non-issue "issue"
mconsidine

**Comment by rlancaste** [comment]
Thank you very much for the feedback.  I will see if I can update documentation and maybe the script to install some things that you were missing.  

An interesting anecdote, I was at our DAS astronomy workshop at Mt. Cuba Observatory this evening and I was talking with Dave Groski for a few minutes about some projects we were working on, and I asked him if he was going to get a Raspberry Pi soon as he had said he was going to do, and he mentioned about a friend in Vermont with a Rock Pi 4 and I put 2 and 2 together and realized that you were the person he was talking about.  Very cool, and it is a small world after all. . .

Rob

**Comment by mconsidine** [comment]
Yes, Dave has been in the loop on this.
I've been able to control my LX200 Classic in all modes - directly wired, hung off a LAN and standalone as a hotspot.  Also got a cheap webcam working through EKOS.
Now to figure out if I can run an Orion TeleTrack AZ-G with it.  Or how to run the relays on a non-goto scope like an EQ3 (i.e. align and point it manually, but then guide from someplace warm.

Matt
PS  After the fact I realized there *could* be an advantage to using a Debian-based OS, as apparently the Arduino folks have Arduino Create for Linux (but Debian-onlly).  And then there's the BeagleBack board with h-bridges and other stuff built in.  But that only has 512MB of RAM and I think could get too easily taxed.  I say all this just as a way of sharing some random thoughts.  The Rock Pi 4 B Plus I think is going to work fine.  I had connected to it over a LAN which was a router hooked up to an antenna, broadcasting the signal to another antenna (through walls) 100ft away, then running the signal though the electrical wiring of the house, to the router ... and then to my laptop on a kitchen table.
